model/common/src/icon4py/model/common/model_options.py

```diff
@@ -50,8 +75,14 @@ def setup_program(
     vertical_sizes = {} if vertical_sizes is None else vertical_sizes
     offset_provider = {} if offset_provider is None else offset_provider
 
+    if isinstance(backend, gtx.backend.Backend):
+        custom_backend = backend
+    else:
+        # customized backend params
+        custom_backend = customize_backend(**backend)
```

Comment:
Maybe just
```suggestion
    if not isinstance(backend, (gtx.backend.Backend, None)):
        backend = customize_backend(**backend)
```

Reply:
Because it's more concise and therefore should be easier to read. (Though the python type checker might disagree.)

Reply:
You mean you don't like the `not`?